Feat/#30/add ui

sewonkimm opened the pull request.

sewonkimm committed:
[Feat] #30 Header UI 버튼 추가

- image sprite 파일에 이미지 추가
- Header에icon image 모두 sprite 형식으로 불러오는 것으로 변경
- Header에 selectBtn 추가

sewonkimm committed:
[Feat] #30 콘텐츠 이동, 삭제 버튼 추가

- image sprite에 아이콘 추가했음
- 안쓰는 이미지 삭제했음
- FolderMore 컴포넌트를 생성해 색상 수정, 폴더 이동, 폴더 삭제 버튼을 추가했음
- MemoMore 컴포넌트를 생성해 메모 이동, 메모 삭제 버튼을 추가했음

JaeHyeonKim19 commented on app/src/components/FolderMore/styles.scss:
```diff
@@ -0,0 +1,85 @@
+
+    width: 140px;
+    height: 52px;
+    margin-right: 1.3rem;
```
margin-right에만 다른 단위를 쓰셨는데 특별한 이유가 있을까요?

JaeHyeonKim19 commented on app/src/components/FolderMore/index.tsx:
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import styles from './styles.scss';
+
+const FolderMore = () => {
```
FolderMore이라는 변수명을 봤을 때 어떤 기능을 하는 컴포넌트인지 정확하게 알기 어려운 것 같아요. 어떻게 생각하시나요?

sewonkimm replied:
Folder 에 대한 부가적 기능을 뭐라고 표현해야할지... more버튼에서 그대로 따와서 FolderMore라고 지었는데 [폴더 삭제, 이름 변경, 색상변경]등을 반영한 변수명으로 아이디어 있을까요?

sewonkimm replied:
**FolderSetting으로 변경**

soyoungjeong commented on app/src/components/FolderMore/index.tsx:
```diff
@@ -0,0 +1,23 @@
+      <div className={styles.moveBtn}>
+        <div className={styles.moveIcon} />
+        폴더 이동
+      </div>
```
'폴더 이동' 기능 말고 '이름 변경' 기능이었던 것 같은데 폴더 이동도 있었나요??

sewonkimm replied:
폴더 이동은 없었습니다! Figma와 수정에 반영하겠습니다~

sewonkimm replied:
#34

JaeHyeonKim19 commented on app/src/components/FolderMore/styles.scss:
```diff
@@ -0,0 +1,85 @@
+  justify-content: center;
+  align-items: center;
+
+  margin-top: 1.5rem;
```
여기도 rem을 사용하셨네요! 이유를 알 수 있을까요?

JaeHyeonKim19 commented on app/src/components/MemoMore/styles.scss:
```diff
@@ -0,0 +1,60 @@
+  justify-content: center;
+  align-items: center;
+
+  margin-top: 1.5rem;
```
여기도 rem 관련해서 궁금하네요!

sewonkimm replied:
rem 은 별 의미가 없이 사용했습니다. rem을 사용하는 것보다 px로 고정시키는게 좋은가요?

soyoungjeong replied:
rem과 px이 함께 쓰이면 글꼴 크기가 바뀔 때 비율이 깨질 수도 있을 것 같아요! 프로젝트 전체에 rem 또는 px로 통일 하는 것이 좋을 것 같은데 px로 통일하시죠!!

JaeHyeonKim19 reviewed (changes requested):
몇 가지 의문사항있어서 리뷰남겼습니다. 확인 부탁드려요.

soyoungjeong commented on app/src/components/Header/styles.scss:
```diff
@@ -15,10 +15,47 @@
 
-.homeBtn {
-  height: 64px;
+.homeBtnDiv {
```
전 보통 class명을 지을때 하이픈이나 언더라인을 많이 써서 어떤 것으로 할지 함께 정하면 좋을 것 같아요 !

sewonkimm replied:
**class 명 지을 때 언더바 사용하기** 📌

sewonkimm replied:
scss 코딩 컨벤션에 class 네이밍 권장에 따라서 언더바를 사용하기로 했습니다.

soyoungjeong reviewed (changes requested):
추가로 memo edit 페이지에서는 Header에 선택 아이콘이 안나오도록 추후에 처리하는 것이 좋을 것 같네요 !

sewonkimm committed:
[Chore] #30 rem 단위 수정

글꼴 크기가 바뀔 때 비율이 변하지 않도록 css사용하던 단위를 rem에서 px로 통일

sewonkimm committed:
[Chore] #30 class명 변경

scss 코딩 컨벤션 class 네이밍 권장에 따라 class명을 camel case에서 underbar 사용으로 변경

sewonkimm committed:
[Chore] #30 FolderMore, MemoMore 컴포넌트명 변경

변수명의 More의 의미가 명확하지 않아 Setting으로 변경

sewonkimm commented:
디자인에 대한 의견을 반영 #34

sewonkimm commented:
MemoEdit 페이지에 저장버튼 추가 #36

sewonkimm committed:
[Chore] #30 UI 수정사항 반영

1. Foler Setting, Memo Setting 페이지의 디자인 변경에 따른 수정사항 반영
2. 버튼 삭제후 이미지형 아이콘으로 통일
3. image sprite에 이미지 추가

sewonkimm committed:
[Chore] #36 save 버튼 추가

soyoungjeong commented on app/src/components/Header/styles.scss:
```diff
@@ -51,6 +51,15 @@
+
+  background-image: url(../../images/icons.png);
+  background-position: -320px -239px;
+}
```
제가 불러서 확인해봤을 때는 저장 버튼이 선택 버튼으로 보입니다!

sewonkimm replied:
네 정상입니다! 일단 display 설정을 none 으로 해놓았습니다.
edit 페이지 기능 구현하면서 header 설정도 바꿔줄 예정입니다.

JaeHyeonKim19 reviewed (approved):
수고하셨어요👍
